integration/tests/access/cohort3/grpc_state_stream_test.go

```diff
@@ -143,99 +158,189 @@ func (s *GrpcStateStreamSuite) TestHappyPath() {
-		}
-	}()
 
-	has := func(events []flow.Event, eventType flow.EventType) bool {
-		for _, event := range events {
-			if event.Type == eventType {
-				return true
-			}
-		}
-		return false
-	}
+			controlANStream, err := rpc.call(s.ctx, sdkClientControlAN, startValue, &executiondata.EventFilter{})
+			s.Require().NoError(err)
+			controlANEvents, controlANErrs, err := SubscribeEventsHandler(s.ctx, controlANStream)
+			s.Require().NoError(err)
 
-	targetEvent := flow.EventType("flow.AccountCreated")
+			testONStream, err := rpc.call(s.ctx, sdkClientTestON, startValue, &executiondata.EventFilter{})
+			s.Require().NoError(err)
+			testONEvents, testONErrs, err := SubscribeEventsHandler(s.ctx, testONStream)
+			s.Require().NoError(err)
 
-	foundANTxCount := 0
-	foundONTxCount := 0
-	r := newResponseTracker()
-	for {
-		select {
-		case err := <-testANErrs:
-			s.Require().NoErrorf(err, "unexpected test AN error")
-		case err := <-controlANErrs:
-			s.Require().NoErrorf(err, "unexpected control AN error")
-		case err := <-testONErrs:
-			s.Require().NoErrorf(err, "unexpected test ON error")
-		case event := <-testANEvents:
-			if has(event.Events, targetEvent) {
-				s.T().Logf("adding access test events: %d %d %v", event.BlockHeight, len(event.Events), event.Events)
-				r.Add(s.T(), event.BlockHeight, "access_test", &event)
-				foundANTxCount++
+			if rpc.generateEvents {
+				// generate events
+				go func() {
+					s.generateEvents(txGenerator, txCount)
+				}()
 			}
-		case event := <-controlANEvents:
-			if has(event.Events, targetEvent) {
-				s.T().Logf("adding control events: %d %d %v", event.BlockHeight, len(event.Events), event.Events)
-				r.Add(s.T(), event.BlockHeight, "access_control", &event)
+
+			has := func(events []flow.Event, eventType flow.EventType) bool {
+				for _, event := range events {
+					if event.Type == eventType {
+						return true
+					}
+				}
+				return false
 			}
-		case event := <-testONEvents:
-			if has(event.Events, targetEvent) {
-				s.T().Logf("adding observer test events: %d %d %v", event.BlockHeight, len(event.Events), event.Events)
-				r.Add(s.T(), event.BlockHeight, "observer_test", &event)
-				foundONTxCount++
+
+			targetEvent := flow.EventType("flow.AccountCreated")
+
+			foundANTxCount := 0
+			foundONTxCount := 0
+			r := newResponseTracker()
+
+			for {
+				select {
+				case err := <-testANErrs:
+					s.Require().NoErrorf(err, "unexpected test AN error")
+				case err := <-controlANErrs:
+					s.Require().NoErrorf(err, "unexpected control AN error")
+				case err := <-testONErrs:
+					s.Require().NoErrorf(err, "unexpected test ON error")
+				case event := <-testANEvents:
+					if has(event.Events, targetEvent) {
+						s.T().Logf("adding access test events: %d %d %v", event.Height, len(event.Events), event.Events)
+						r.Add(s.T(), event.Height, "access_test", &event)
+						foundANTxCount++
+					}
+				case event := <-controlANEvents:
```

Comment:
can you include a check that the `MessageIndex` provided are sequential? You only need to check for the control responses, since the test responses are compared against them later.

Reply:
[97fd28d](https://github.com/onflow/flow-go/pull/5602/commits/97fd28d5c484117656a98da0ee521fe8dc693552)